component/src/main/java/me/androidbox/component/general/PhotoPicker.kt

```diff
@@ -0,0 +1,197 @@
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil.compose.AsyncImage
+import me.androidbox.component.R
+import me.androidbox.component.ui.theme.BusbyTaskyTheme
+import me.androidbox.component.ui.theme.photoBackgroundColor
+import me.androidbox.component.ui.theme.photoPickerBorderColor
+import me.androidbox.component.ui.theme.photoTextColor
+
+@Composable
+fun PhotoPicker(
+    listOfPhotoUri: List<Uri>,
+    onPhotoUriSelected: (photoUri: Uri) -> Unit,
+    modifier: Modifier = Modifier) {
+
+    val photoPickerLauncher = rememberLauncherForActivityResult(
+        contract = ActivityResultContracts.PickVisualMedia(),
+        onResult = { uri ->
+            if (uri != null) {
+                Log.d("PHOTO", "Path of image [ $uri ]")
+                onPhotoUriSelected(uri)
+            }
+        })
+
+    /* We have no images attached */
+    if(listOfPhotoUri.isEmpty()) {
+        AddFirstPhoto(modifier = modifier) {
+            photoPickerLauncher.launch(PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly))
+        }
+    }
+    else {
+        AddSequentialPhoto(modifier = modifier, selectedImageUri = listOfPhotoUri, onAddPhotosClicked = {
+            photoPickerLauncher.launch(PickVisualMediaRequest(ActivityResultContracts.PickVisualMedia.ImageOnly))
+        })
+    }
+}
+
+@Composable
+fun AddFirstPhoto(
+    modifier: Modifier,
+    onAddPhotosClicked: () -> Unit) {
+    Row(modifier = modifier
+        .height(100.dp),
+        horizontalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically) {
+
+        IconButton(onClick = {
+            onAddPhotosClicked()
+        }) {
+            Image(
+                painter = painterResource(id = R.drawable.add_photo),
+                contentDescription = stringResource(id = R.string.add_photos)
+            )
+        }
+
+        Spacer(modifier = Modifier.width(14.dp))
+        Text(
+            text = stringResource(R.string.add_photos),
+            fontWeight = FontWeight.SemiBold,
+            fontSize = 16.sp,
+            color = MaterialTheme.colorScheme.photoTextColor
+        )
+    }
+}
+
+@Composable
+fun AddSequentialPhoto(modifier: Modifier, selectedImageUri: List<Uri>, onAddPhotosClicked: () -> Unit) {
+    /* We have images that have been selected */
+    Column(modifier = modifier) {
+        Spacer(modifier = Modifier.height(20.dp))
+
+        Text(
+            text = stringResource(R.string.photos),
+            fontSize = 20.sp,
+            fontWeight = FontWeight.SemiBold
+        )
+
+        Spacer(modifier = Modifier.height(20.dp))
+
+        LazyRow(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = Arrangement.spacedBy(12.dp),
+            state = rememberLazyListState()) {
```

Comment:
There's no benefit of providing such a state if you don't do something with it. Each LazyRow already contains this as a default parameter

Reply:
@philipplackner Removed this rememberLazyListState if it already has this as its default component